itemManagement/views.py

```diff
@@ -37,15 +40,67 @@ def get_context_data(self, *args, **kwargs):
 class ItemDetailsView(TemplateView):
     def get(self, request, itemId, *args, **kwargs):
         isAdmin = request.user.is_superuser
+        # isAdmin = True
 
         print(f"Item ID requested: {itemId}")
 
         context = Item.objects.get(id=itemId).getItemDetails()
+
         if isAdmin:
             return render(request, 'itemManagement/item_details_admin.html', context=context)
         else:
             return render(request, 'itemManagement/item_details_assistant.html', context=context)
 
+    def post(self, request, itemId, *args, **kwargs):
+
+        item = Item.objects.get(id=itemId)
+
+        item.lastUsed = timezone.now()
+        # TODO: item lastUsed updated if decrement clicked
+
+        isAdmin = True
+
+        # TODO: Admin validation
+
+        # Admin Fields updating fields other than quantities if admin
+        if isAdmin:
+            # --------TEXT FIELDS--------
+            item.title = request.POST.get('itemName', "").strip()
+            item.description = request.POST.get('description', "").strip()
+            item.price = request.POST.get('price', "").strip()
+            item.unit = request.POST.get('unit', "").strip()
+            item.save(update_fields=['title', 'description', 'price', 'unit'])
+            # ---------------------------
+
+            # --------TAGS-------------
+            newTags = request.POST.get('newTags', "").split(',')
+            # clear all tags
+            item.tag_set.all().delete()
+            # add back all new edited tags
+            for newTag in newTags:
+                if newTag != "" or newTag is None:
+                    newTag = Tag.objects.create(name=newTag.strip(), item=item)
+                    newTag.save()
+            # -------------------------
+
+        # ----Quantities in Locations----
+        # Locations: list of all locations that house the current item
+        locations = item.locations.all()
+
+        # Locations
+        for location in locations:
+            itemStorage = location.itemstorage_set.get(item=item)
+            itemStorage.quantity = request.POST.get('quantity-location-' + str(location.id), "").strip()
+            itemStorage.save()
+        # -------------------------------
+
+        # TODO: Post Images
+
+        # TODO: Input validation
+
+        # TODO: Return to homepage with same previous state after POST
+        return HttpResponse(status=204)
```

Comment:
Just checking, does this resubmit the form on refresh (we probably don't want that to happen if it does)?

Reply:
The request is done by ajax, so no

Reply:
Are we doing any error handling?

Reply:
No, that's in my todo. I guess "Input validation" is kinda not the right word but that's what that meant.

Reply:
Ok, just wanted to make sure its still planned and thought of

Reply:
Appreciate it either way